### PR TITLE
fix: delete methods not resolving #161

### PR DESCRIPTION
**Pull request by matinzd**

### CODE REVIEW

File: android/src/main/java/dev/matinzd/healthconnect/HealthConnectManager.kt [hunk 1]
```diff
@@ -255,12 +255,17 @@ class HealthConnectManager(private val applicationContext: ReactApplicationConte
...
-            recordType = record,
-            recordIdsList = recordIdsList.toArrayList().mapNotNull { it.toString() }.toList(),
-            clientRecordIdsList = if (clientRecordIdsList.size() > 0) clientRecordIdsList.toArrayList()
-              .mapNotNull { it.toString() }.toList() else emptyList()
-          )
+          try {
+            healthConnectClient.deleteRecords(
+              recordType = record,
+              recordIdsList = recordIdsList.toArrayList().mapNotNull { it.toString() }.toList(),
+              clientRecordIdsList = if (clientRecordIdsList.size() > 0) clientRecordIdsList.toArrayList()
```
Comment (TheRogue76) [inline android/src/main/java/dev/matinzd/healthconnect/HealthConnectManager.kt:262]: Why do you need to check the size? If it is an empty list, the rest of these should run just fine

**Review by TheRogue76 — COMMENTED** [review]
Left two comments

File: android/src/main/java/dev/matinzd/healthconnect/HealthConnectManager.kt (hunk 1 above)
Comment (matinzd) [reply android/src/main/java/dev/matinzd/healthconnect/HealthConnectManager.kt:262]: Yeah makes sense. I am gonna remove the check.

Comment (matinzd) [reply android/src/main/java/dev/matinzd/healthconnect/HealthConnectManager.kt]: Done [99cca4d](https://github.com/matinzd/react-native-health-connect/pull/177/commits/99cca4dacd999e7663c203ffe828fb8300b82b57).

File: android/src/main/java/dev/matinzd/healthconnect/HealthConnectManager.kt [hunk 2]
```diff
@@ -255,12 +255,17 @@ class HealthConnectManager(private val applicationContext: ReactApplicationConte
...
+                .mapNotNull { it.toString() }.toList() else emptyList()
+            )
+            promise.resolve(true)
+          } catch (e: Exception) {
```
[the remark this answers is not in the document]
Comment (TheRogue76) [reply android/src/main/java/dev/matinzd/healthconnect/HealthConnectManager.kt:265]: https://kotlinlang.org/docs/exceptions.html#exception-hierarchy. Technically, Exception is only a subclass, and Throwable is the base class. Guessing this API on Google's side is not so brain dead to be throwing an Error is it?

**Review by TheRogue76 — APPROVED** [review]

File: android/src/main/java/dev/matinzd/healthconnect/HealthConnectManager.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (matinzd) [reply android/src/main/java/dev/matinzd/healthconnect/HealthConnectManager.kt:265]: If I want to make changes to this, I need to fix it in other places. I will merge this for now and change the class later on in another PR.

**Pull request merged**